### PR TITLE
Add Minio support.

### PR DESCRIPTION
**Pull request by davinchia**
## What
* #4206 
* Enable logging to Minio.
* Enable reading from Minio.
* Deploy minio as part of the Kubernetes dev and stable deploys.
* Documentation (will follow up and tag Abhi)

Also make log level injectable for easier debugging in the future. This defaults to INFO.

## How
* Configure the logger to accept a minio endpoint. Use `s3PathStyleAccess` is required for this.
* Pass this endpoint to S3 client so it can read from minio.
* Add a minio deploy as part of the base overlay. Configure so this is always deployed.

## Pre-merge Checklist
Expand the checklist which is relevant for this PR. 

<details><summary> <strong> Connector checklist </strong> </summary>
<p>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] Secrets are annotated with `airbyte_secret` in the connector's spec
- [ ] Credentials added to Github CI if needed and not already present. [instructions for injecting secrets into CI](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#using-credentials-in-ci). 
- [ ] Unit & integration tests added as appropriate (and are passing)
    * Community members: please provide proof of this succeeding locally e.g: screenshot or copy-paste acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] `/test connector=connectors/<name>` command as documented [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-an-existing-connector) is passing. 
    * Community members can skip this, Airbyters will run this for you. 
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] `README.md`
    - [ ] `docs/SUMMARY.md` if it's a new connector
    - [ ] Created or updated reference docs in `docs/integrations/<source or destination>/<name>`.
    - [ ] Changelog in the appropriate page in `docs/integrations/...`. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md` contains a reference to the new connector
    - [ ] Build status added to [build page](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/builds.md)
- [ ] Build is successful
- [ ] Connector version bumped like described [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-a-connector)
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-a-connector)
- [ ] No major blockers
- [ ] PR merged into master branch
- [ ] Follow up tickets have been created
- [ ] Associated tickets have been closed & stakeholders notified
</p>
</details>

<details><summary> <strong> Connector Generator checklist </strong> </summary>
<p>
   
- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed.
</p>
</details>

### CODE REVIEW

**Comment by davinchia** [comment]
@jrhizor what do you think about this + automatically deploying with test airbyte and leaving prod airbyte empty?

If this makes sense I'll also add documentation on how to use Minio with production airbyte.

**Comment by jrhizor** [comment]
I think we will want a prod overlay that uses minio so there's a one apply deploy people can use to get a fully working instance on a stable version in their cluster. I would bet that many users would prefer this as the "default" option there and allow the extensibility to configure a different backing persistence. 

We'll have to figure out how to document this well though and explain the benefits of customizing the configuration.

The implementation pieces of this PR look good though!

File: kube/overlays/dev/standalone-minio.yaml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  labels:
+    app: minio-storage-claim
+spec:
+  # Read more about access modes here: http://kubernetes.io/docs/user-guide/persistent-volumes/#access-modes
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    # This is the request for storage. Should be available in the cluster.
+    requests:
+      storage: 10Mi
```
Comment (jrhizor) [inline kube/overlays/dev/standalone-minio.yaml:59]: needs to be much higher

File: kube/overlays/dev/standalone-minio.yaml [hunk 2]
```diff
@@ -0,0 +1,71 @@
+apiVersion: apps/v1 #  for k8s versions before 1.9.0 use apps/v1beta2  and before 1.8.0 use extensions/v1beta1
+kind: Deployment
+metadata:
+  # This name uniquely identifies the Deployment
+  name: minio-deployment
```
Comment (jrhizor) [inline kube/overlays/dev/standalone-minio.yaml:5]: all the minio deployments/pods/service should be scoped by airbyte

**Comment by davinchia** [comment]
@jrhizor addressed all the comments. I was only able to test `stable` deploy with dev images.

I will write documentation in a separate PR.

File: airbyte-commons/src/main/resources/log4j2.xml [hunk 3]
```diff
@@ -3,12 +3,14 @@
     <Properties>
         <Property name="default-pattern">%d{yyyy-MM-dd HH:mm:ss}{GMT+0} %highlight{%p} %C{1.}(%M):%L - %X - %m%n</Property>
         <Property name="default-worker-file-pattern">%d{yyyy-MM-dd HH:mm:ss}{GMT+0} %p (%X{job_root}) %C{1}(%M):%L - %m%n</Property>
-
+        <!-- Always log INFO by default. -->
+        <Property name="log-level">$${env:LOG_LEVEL:-INFO}</Property>
```
Comment (davinchia) [inline airbyte-commons/src/main/resources/log4j2.xml:7]: @jrhizor this is the default stuff you were talking about

**Pull request merged**